utils.py

```diff
@@ -39,103 +51,643 @@ def get_colors(**kwargs):
+    create_index = ElasticCreateSiren(
+        elastic_url=ELASTIC_URL,
+        elastic_index=elastic_index,
+        elastic_user=ELASTIC_USER,
+        elastic_password=ELASTIC_PASSWORD,
+        elastic_bulk_size=ELASTIC_BULK_SIZE,
+    )
+    create_index.execute()
+
+
+def fill_elastic_index(**kwargs):
+    next_color = kwargs["ti"].xcom_pull(key="next_color", task_ids="get_colors")
+    elastic_index = "siren-" + next_color
+    siren_db_conn, siren_db_cursor = connect_to_db()
+    siren_db_cursor.execute(f'''
+        SELECT 
+            ul.siren,
+            st.siret as siret_siege,
+            st.date_creation as date_creation_siege,
+            st.tranche_effectif_salarie as tranche_effectif_salarie_siege,
+            st.date_debut_activite as date_debut_activite_siege,
+            st.etat_administratif_etablissement as etat_administratif_siege,
+            st.activite_principale as activite_principale_siege,
+            st.complement_adresse as complement_adresse,
+            st.numero_voie as numero_voie,
+            st.indice_repetition as ndice_repetition,
+            st.type_voie as type_voie,
+            st.libelle_voie as libelle_voie,
+            st.distribution_speciale as distribution_speciale,
+            st.cedex as cedex,
+            st.libelle_cedex as libelle_cedex,
+            st.commune as commune,
+            st.libelle_commune as libelle_commune,
+            st.code_pays_etranger as code_pays_etranger,
+            st.libelle_commune_etranger as libelle_commune_etranger,
+            st.libelle_pays_etranger as libelle_pays_etranger,
+            st.code_postal as code_postal,
+            st.geo_id as geo_id,
+            st.longitude as longitude,
+            st.latitude as latitude,
+            st.activite_principale_registre_metier as activite_principale_registre_metier,
+            st.adresse_complete as adresse_complete,
+            ul.date_creation_unite_legale as date_creation_unite_legale,
+            ul.tranche_effectif_salarie_unite_legale as tranche_effectif_salarie_unite_legale,
+            ul.date_mise_a_jour_unite_legale as date_mise_a_jour,
+            ul.categorie_entreprise as categorie_entreprise,
+            ul.etat_administratif_unite_legale as etat_administratif_unite_legale,
+            ul.nom_raison_sociale as nom_raison_sociale,
+            ul.nature_juridique_unite_legale as nature_juridique_unite_legale,
+            ul.activite_principale_unite_legale as activite_principale_unite_legale,
+            ul.economie_sociale_solidaire_unite_legale as 
+            economie_sociale_solidaire_unite_legale,
+            (SELECT count FROM count_etab ce WHERE ce.siren = st.siren) as nombre_etablissements,
+            (SELECT count FROM count_etab_ouvert ceo WHERE ceo.siren = st.siren) as nombre_etablissements_ouverts,
+            (SELECT liste_enseignes FROM enseignes le WHERE le.siren = st.siren) as liste_enseignes,
```

Comment:
something like (not tested) : 
```
(SELECT json_group_array(
                json_object(
                    \'enseigne_1\', enseigne_1,
                    \'enseigne_2\', enseigne_2,
                    \'enseigne_3\', enseigne_3
                    )
                ) FROM 
                (
                    SELECT enseigne_1, enseigne_2, enseigne_3 from siret
                    WHERE siren = st.siren
                )
            ) as liste_enseignes
```

Reply:
then you need to process it in a python function

Reply:
Yes, I used the same logic!